src/wallet/wallet.cpp

```diff
@@ -4112,8 +4119,10 @@ util::Result<void> CWallet::ApplyMigrationData(WalletBatch& local_wallet_batch,
         if (data.master_key.key.IsValid()) {
             SetupDescriptorScriptPubKeyMans(local_wallet_batch, data.master_key);
         } else {
-            // Setup with a new seed if we don't.
-            SetupOwnDescriptorScriptPubKeyMans(local_wallet_batch);
+            // Setup with a new seed, only if we are actually migrating any spendable descriptor
+            if (!empty_local_wallet) {
```

Comment:
In 37e3cab479080db9a3ac10f9b9b5c7f6c0e07882 "wallet: migration, don't create spendable wallet from a watch-only legacy wallet"

This check can go above next to the `!IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS)`.

Reply:
sure, done as suggested. Also added an extra `!data->master_key.key.IsValid()` to the `empty_local_wallet` definition.